cloud/lib/ui-stack.ts

```diff
@@ -0,0 +1,225 @@
+		*/
+		const albOrigin = new HttpOrigin(apiDomainName, {
+			customHeaders: {
+				[customAuthHeaderName]: customAuthHeaderValue,
+			},
+		});
+		const siteDistribution = new Distribution(this, generateResourceId('site-distribution'), {
+			defaultRootObject: 'index.html',
+			certificate,
+			domainNames: [domainName],
+			enableLogging: true,
+			errorResponses: [
+				{
+					httpStatus: 404,
+					responseHttpStatus: 200,
+					responsePagePath: '/index.html',
+					ttl: Duration.seconds(30),
+				},
+			],
+			defaultBehavior: {
+				origin: new S3Origin(hostBucket, {
+					originAccessIdentity: cloudfrontOAI,
+				}),
+				cachePolicy: new CachePolicy(this, generateResourceId('site-cache-policy'), {
+					cookieBehavior: CacheCookieBehavior.allowList(`${appName}.sid`),
+				}),
+				compress: true,
+				allowedMethods: AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
+				viewerProtocolPolicy: ViewerProtocolPolicy.HTTPS_ONLY,
+			},
+			additionalBehaviors: {
+				'/api/documents/*': {
+					// Cache static content, currently just docs
+					origin: albOrigin,
+					cachePolicy: new CachePolicy(this, generateResourceId('static-resources'), {
+						comment: 'Long-lived cache for static files, invalidation forced after redeploy',
+						minTtl: Duration.days(14),
+					}),
+					viewerProtocolPolicy: ViewerProtocolPolicy.HTTPS_ONLY,
+					originRequestPolicy: OriginRequestPolicy.ALL_VIEWER_EXCEPT_HOST_HEADER,
+					responseHeadersPolicy: ResponseHeadersPolicy.SECURITY_HEADERS,
+					edgeLambdas: [
+						{
+							functionVersion: verifierEdgeFunction.currentVersion,
+							eventType: LambdaEdgeEventType.VIEWER_REQUEST,
+						},
+					],
+				},
+				'/api/*': {
+					// Treat all other paths as dynamic, do not cache
+					origin: albOrigin,
+					allowedMethods: AllowedMethods.ALLOW_ALL,
+					cachePolicy: CachePolicy.CACHING_DISABLED,
+					viewerProtocolPolicy: ViewerProtocolPolicy.HTTPS_ONLY,
+					originRequestPolicy: OriginRequestPolicy.ALL_VIEWER_EXCEPT_HOST_HEADER,
+					responseHeadersPolicy: ResponseHeadersPolicy.SECURITY_HEADERS,
+					edgeLambdas: [
+						{
+							functionVersion: verifierEdgeFunction.currentVersion,
+							eventType: LambdaEdgeEventType.VIEWER_REQUEST,
+						},
+					],
+				},
+			},
+			priceClass: PriceClass.PRICE_CLASS_100,
+		});
+		this.cloudFrontDistributionId = new CfnOutput(this, 'CloudFrontDistributionId', {
+			value: siteDistribution.distributionId,
+			exportName: `${this.stackName}-CloudFrontDistributionId`,
+		});
+
+		/*
+		DNS Records for Route53
+		*/
+		const websiteTarget = RecordTarget.fromAlias(new CloudFrontTarget(siteDistribution));
+		new ARecord(this, generateResourceId('arecord-cfront'), {
+			zone: hostedZone,
+			recordName: domainName,
+			target: websiteTarget,
+			deleteExisting: true,
+			comment: `DNS A Record for UI host (${stage})`,
+		});
+		new AaaaRecord(this, generateResourceId('aaaarecord-cfront'), {
```

Comment:
What's the `AaaaRecord` used for?

Reply:
IPv6 access. CF distributions are dual stack by default, and I saw no reason to remove that.